src/nlp.jl

```diff
@@ -619,7 +617,39 @@ function MOI.eval_constraint_jacobian(d::NLPEvaluator, J, x)
     return
 end
 
+function MOI.eval_constraint_jacobian_product(d::NLPEvaluator,
+                                              Jv::AbstractVector{Float64},
```

Comment:
If you're going to use one-letter variable names, then it's better to use the ones that are consistent with the MOI documentation: http://www.juliaopt.org/MathOptInterface.jl/stable/apireference/#MathOptInterface.eval_constraint_jacobian_product
i.e., `eval_constraint_jacobian_product(d, y, x, w)`. Same comment for below.